src/main/java/mate/academy/spring/security/CustomUserDetailService.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.spring.security;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String userName) throws UsernameNotFoundException {
+        User user = userService.findByEmail(userName)
+                .orElseThrow(() -> new UsernameNotFoundException(
+                        "User: " + userName + " not found"));
+        UserBuilder builder =
+                org.springframework.security.core.userdetails.User.withUsername(userName);
+        builder.password(user.getPassword());
+        builder.authorities(user.getRoles().stream()
```

Comment:
```suggestion
        builder.roles(user.getRoles().stream()
```
we talked about the difference today